components/service/fxa/src/main/java/mozilla/components/fxa/FirefoxAccount.kt

```diff
@@ -0,0 +1,133 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.fxa
+
+import android.text.TextUtils
+import android.util.Log
+
+class FirefoxAccount(override var rawPointer: FxaClient.RawFxAccount?) : RustObject<FxaClient.RawFxAccount>() {
+
+    constructor(config: Config, clientId: String): this(null) {
+        val e = Error.ByReference()
+        val result = FxaClient.INSTANCE.fxa_new(config.consumePointer(), clientId, e)
+        if (e.isSuccess()) {
+            this.rawPointer = result
+        } else {
+            Log.e("FirefoxAccount.init", e.consumeMessage())
+            this.rawPointer = null
+        }
```

Comment:
All those checks make me wonder if we could turn Error into a sealed class with distinct types. With that we could just do something like:

```Kotlin
when (result) {
    is Success -> ... // Kotlin will do a nice smart cast here
    is ...
}
```